### PR TITLE
Fix bug where all errors were surfaced as workflow-level errors

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
https://github.com/aqueducthq/aqueduct/pull/1107 introduced a bug where all errors, includes those that happen within an operator's context, were surfaced as workflow-level, which meant that every failing workflow got a banner.

This PR does the following:
1) Fixes the immediate error, so that workflow-level errors only occur when the workflow doesn't have an op failure exception.
2) In the process of fixing the above bug, I noticed that our error comparison method was only comparing the top-level errors. I think for our error comparisons, we should be completely unwrapping our candidate error and matching it against the target error. It doesn't make sense to compare just agains the outermost layer since it's much harder to control how much an error is wrapped as it works it way out of our system. I thiiink what really matters is the innermost error, so that's the one we should be comparing. @saurav-c curious what you think? 

Also tested that workfow timeout errors are appropriately surfaced as workflow-level errors:
image

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Review by cw75 — COMMENTED** [review]
@kenxu95 Looks good overall, left a couple comments.

File: src/golang/lib/errors/errors.go [hunk 1]
```diff
@@ -4,23 +4,38 @@ import (
 	"github.com/dropbox/godropbox/errors"
 )
 
-// Is reports whether err matches target
-func Is(err, target error) bool {
-	dboxErr, isErrDbox := err.(errors.DropboxError)
-	dboxTarget, isTargetDbox := target.(errors.DropboxError)
+// Fetches the error message from the error. For Dropbox errors, we exclude the stack trace.
+func getMessage(err error) string {
```
[the remark this answers is not in the document]
Comment (cw75) [reply src/golang/lib/errors/errors.go:8]: I remember that method accounts for non-dropbox errors as well.

[the remark this answers is not in the document]
Comment (cw75) [reply src/golang/lib/errors/errors.go:8]:
```
func GetMessage(err interface{}) string {
	switch e := err.(type) {
	case DropboxError:
		return extractFullErrorMessage(e, false)
	case runtime.Error:
		return runtime.Error(e).Error()
	case error:
		return e.Error()
	default:
		return "Passed a non-error to GetMessage"
	}
}
```

[the remark this answers is not in the document]
Comment (kenxu95) [reply src/golang/lib/errors/errors.go:8]: Oh there's a GetMessage() on Dbxerror and also a GetMessage on the errors library. Got it, will fix.

**Review by cw75 — APPROVED** [review]

**Pull request merged**